src/SessionOne.java

```diff
@@ -5,50 +5,73 @@ public static void main(String[] args) {
-                    isAnStrongNumber(number);
-            }
+        switch (option) {
+            case 1:
+                isAnAmstrongNumber(number);
+            case 2:
+                isAnStrongNumber(number);
+            case 3:
+                isAPerfectNumber(Integer.valueOf(number));
         }
+    }
 
-        public static void isAnAmstrongNumber(String number) {
-            String[] digits = number.split("(?<=.)");
-            double is_amstrong = 0;
-            for (int i = 0; i < digits.length; i++) {
-                is_amstrong += Math.pow(Double.parseDouble(digits[i]), 3);
-            }
-            if (Integer.valueOf(number) == is_amstrong) {
-                System.out.println("it is an SessionOne number");
-            } else {
-                System.out.println("it is not an SessionOne number");
-            }
+    public static void isAnAmstrongNumber(String number) {
+        String[] digits = number.split("(?<=.)");
+
+        double is_amstrong = 0;
+
+        for (int i = 0; i < digits.length; i++) {
+            is_amstrong += Math.pow(Double.parseDouble(digits[i]), 3);
         }
 
-        public static void isAnStrongNumber(String number){
-            String[] digits = number.split("(?<=.)");
-
-            double isStrong = 0;
-            for (int i = 0; i < digits.length; i++) {
-                int factorial = 1;
-                int digit = Integer.valueOf(digits[i]);
-                for (int j = digit; j > 1; j=j-2) {
-                    int tempNumber = 0;
-                    tempNumber = (j * (j-1));
-                    factorial *= tempNumber;
-                }
-                isStrong +=factorial;
-            }
-            if(isStrong == Integer.valueOf(number)){
-                System.out.println("The number is a Strong number "+isStrong);
+        if (Integer.valueOf(number) == is_amstrong) {
+            System.out.println("it is an SessionOne number");
+        } else {
+            System.out.println("it is not an SessionOne number");
+        }
+    }
+
+    public static void isAnStrongNumber(String number) {
+        String[] digits = number.split("(?<=.)");
+
+        double isStrong = 0;
+
+        for (int i = 0; i < digits.length; i++) {
+            int factorial = 1;
+            int digit = Integer.valueOf(digits[i]);
+
+            for (int j = digit; j > 1; j = j - 2) {
+                int tempNumber = 0;
+                tempNumber = (j * (j - 1));
+                factorial *= tempNumber;
             }
-            else{
-                System.out.println("The number is not a Strong number "+isStrong);
+            isStrong += factorial;
+        }
+
+        if (isStrong == Integer.valueOf(number)) {
+            System.out.println("The number is a Strong number " + isStrong);
+        } else {
+            System.out.println("The number is not a Strong number " + isStrong);
+        }
+    }
+
+    public static void isAPerfectNumber(int number) {
```

Comment:
I think all these functions are good candidates to be tested, you could write some good unit tests for them, functions are simple, so seems like a good opportunity to explore the basics of it. Think about it. Here's a tutorial in a quick google search, covers some of the bassics: [Testing with JUnit](https://www.vogella.com/tutorials/JUnit/article.html)

Reply:
I will, at the end of this unit. I promise. I don't even finish the exercises for the first week. OMG

Reply:
"I haven't even finished"* or "I didn't even finished"

Yes, it's a suggestion, you are building your profile as a Quality Assurance Engineer, so why not to know about unit testing as well, it's part of it in a way, if at the end you move to development, it will be useful too.

Also, take your time, don't rush your pace, remember that "el que abarca much, few aprieta" this is important, if you overwhelm yourself studying 3 or 4 things at the same time, you will burn out, learn to prioritize, focus on 2 at most, set a goal, finish it, and only then move on. There's no rush and the more you learn, the more you'll discover things that you would like to learn. Choose wisely and commit to it. Otherwise you'll get frustrated. If you keep a steady peace, you'll move forward and you will make progress, believe on that (Think in your process, all the study that you have been doing is paying off, you know a lot more than before, just continue that way) Remember that studying is not the only thing in your life right now, learn to balance, without losing focus. Keep it up, this is going good.

Reply:
Actually, I have seen that in some positions. "Unit testing".

Reply:
> I think all these functions are good candidates to be tested, you could write some good unit tests for them, these functions are really simple, so seems like a good opportunity to explore the basics of it. Think about it. Here's a tutorial I found after doing a quick google search, covers some of the bassics: [Testing with JUnit](https://www.vogella.com/tutorials/JUnit/article.html)

you know... unit testing was part of the workshop exercise, it seems kind of cool 😸

Reply:
You should start doing it while you continue with these exercises, and using TDD as the methodology. Then you will understand the point of them. Test shouldn't be left to be done at the end in this type of scenario.